pkg/segment/structs/evaluationstructs.go

```diff
@@ -771,10 +908,16 @@ func (self *ConditionExpr) EvaluateCondition(fieldToValue map[string]utils.CValu
 }
 
 func (self *TextExpr) GetFields() []string {
+	var fields []string
 	if self.IsTerminal || (self.Op != "max" && self.Op != "min") {
-		return self.Value.GetFields()
+		if self.Value != nil {
+			fields = append(fields, self.Value.GetFields()...)
+		}
+		if self.Val != nil {
+			fields = append(fields, self.Val.GetFields()...)
+		}
+		return fields
 	}
```

Comment:
Don't you need to call `GetFields()` on all the fields you added to the `TextExpr`?

Reply:
Yes made the changes